File: frontend/src/api/Book.ts
import request from "@/utils/request";

export interface Book {
    bookId: number,
    title: string,
    author: string,
    isbn: string,
    categoryId: number,
    publisher: string,
    publishDate: string,
    totalStock: number,
    availableStock: number,
    summary: string,
    totalLend: number,
    status: number,
    createTime: string,  
}

export const  getTop10 = () => {
    return request({
        url: `/user/book/top10`,
        method: 'GET',
    })
}

export const getTop10ByCategory = (categoryId: number) => {
    return request({
        url: `/user/book/top10ByCategory/${categoryId}`,
        method: 'GET',
    })
}

export const getBookById = (bookId: number) => {
    return request({
        url: `/user/book/getBookById/${bookId}`,
        method: 'GET',
    })
}
export const getNewBooks = () => {
    return request({
        url: `/user/book/getNewBooks`,
        method: 'GET',
    })
}

export const searchBooks = (keyword: string) => {
    return request({
        url: `/user/book/searchBooks`,
        method: 'GET',
        params: {
            keyword
        }
    })
}

File: backend/src/main/java/com/x/pojo/entity/Book.java
package com.x.pojo.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Book {
    private Integer bookId;          // 图书ID
    private String title;           // 书名
    private String author;          // 作者
    private String isbn;            // ISBN
    private Integer categoryId;     // 分类ID (外键)
    private String publisher;       // 出版社
    private LocalDate publishDate;  // 出版日期
    private Integer totalStock;     // 总库存
    private Integer availableStock;
    private String summary;         // 摘要
    private Long totalLend ;        // 总借阅
    private Byte status;            // 状态 (0=正常，1=下架)
    private LocalDateTime createTime;
}
